cluster-autoscaler/cloudprovider/clusterapi/README.md

```diff
@@ -0,0 +1,56 @@
+# Cluster Autoscaler on Cluster API
+
+The cluster autoscaler on [Cluster API](https://cluster-api.sigs.k8s.io/) uses
+the [cluster-api project](https://github.com/kubernetes-sigs/cluster-api) to
+manage the provisioning and de-provisioning of nodes within a Kubernetes
+cluster.
+
+## Kubernetes Version
+
+The cluster-api provider requires Kubernetes v1.16 or greater to run the
+v1alpha3 version of the API.
+
+## Starting the Autoscaler
+
+To enable the Cluster API provider, you must first specify it in the command
+line arguments to the cluster autoscaler binary. For example:
+
+```
+cluster-autoscaler --cloud-provider=clusterapi
+```
+
+Please note, this example only shows the cloud provider options, you will
+most likely need other command line flags. For more information you can invoke
+`cluster-autoscaler --help` to see a full list of options.
+
+## Enabling Autoscaling
+
+To enable the automatic scaling of components in your cluster-api managed
+cloud there are a few annotations you need to provide. These annotations
+must be applied to either [MachineSet](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine-set.html)
+or [MachineDeployment](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine-deployment.html)
+resources depending on the type of cluster-api mechanism that you are using.
+
+There are two annotations that control how a cluster resource should be scaled:
+
+* `cluster.k8s.io/cluster-api-autoscaler-node-group-min-size` - This specifies
```

Comment:
may be mention setting this to `0` is not currently supported.

Reply:
i mention it below, but perhaps it would be clearer if i just put that sentence here.

Reply:
The reasoning is because we don't support scaling from zero yet, we won't ever set expected replicas < 1.

Reply:
yeah, agreed. i'll move it.